sections/collage.liquid

```diff
@@ -31,15 +31,15 @@
                 <div class="media media--transparent ratio"{% if block.settings.image != blank %} style="--ratio-percent: {{ 1 | divided_by: block.settings.image.aspect_ratio | times: 100 }}%"{% else %} style="--ratio-percent: 100%"{% endif %}>
                   {%- if block.settings.image != blank -%}
                     <img
-                      srcset="{%- if block.settings.image.width >= 550 -%}{{ block.settings.image | img_url: '550x' }} 550w,{%- endif -%}
-                        {%- if block.settings.image.width >= 720 -%}{{ block.settings.image | img_url: '720x' }} 720w,{%- endif -%}
-                        {%- if block.settings.image.width >= 990 -%}{{ block.settings.image | img_url: '990x' }} 990w,{%- endif -%}
-                        {%- if block.settings.image.width >= 1100 -%}{{ block.settings.image | img_url: '1100x' }} 1100w,{%- endif -%}
-                        {%- if block.settings.image.width >= 1500 -%}{{ block.settings.image | img_url: '1500x' }} 1500w,{%- endif -%}
-                        {%- if block.settings.image.width >= 2200 -%}{{ block.settings.image | img_url: '2200x' }} 2200w,{%- endif -%}
-                        {%- if block.settings.image.width >= 3000 -%}{{ block.settings.image | img_url: '3000x' }} 3000w,{%- endif -%}
-                        {{ block.settings.image | img_url: 'master' }} {{ block.settings.image.width }}w"
-                      src="{{ block.settings.image | img_url: '1500x' }}"
+                      srcset="{%- if block.settings.image.width >= 550 -%}{{ block.settings.image | image_url: width: 550 }} 550w,{%- endif -%}
+                        {%- if block.settings.image.width >= 720 -%}{{ block.settings.image | image_url: width: 720 }} 720w,{%- endif -%}
+                        {%- if block.settings.image.width >= 990 -%}{{ block.settings.image | image_url: width: 990 }} 990w,{%- endif -%}
+                        {%- if block.settings.image.width >= 1100 -%}{{ block.settings.image | image_url: width: 1100 }} 1100w,{%- endif -%}
+                        {%- if block.settings.image.width >= 1500 -%}{{ block.settings.image | image_url: width: 1500 }} 1500w,{%- endif -%}
+                        {%- if block.settings.image.width >= 2200 -%}{{ block.settings.image | image_url: width: 2200 }} 2200w,{%- endif -%}
+                        {%- if block.settings.image.width >= 3000 -%}{{ block.settings.image | image_url: width: 3000 }} 3000w,{%- endif -%}
+                        {{ block.settings.image | image_url }} {{ block.settings.image.width }}w"
```

Comment:
Also, if this is the case, then `theme-check` should probably automatically convert `master` to `5760` when autocorrecting.

Reply:
>The intention is that folks shouldn't use it without width or height. While omitting the param does the same thing, it's a bad default because it means the page performance depends on merchant-uploaded-assets. And if you are a theme dev and guarantee some form of perf, that's no good.

Thanks for the context @charlespwd - much appreciated!

@tyleralsbury I'm good with using this undocumented feature for the time being as long as the [Size parameters for `image_url`](https://shopify.dev/api/liquid/filters/url-filters#size-parameters) doc is updated to reflect this:

>The `image_url` filter requires either a `width` or `height` attribute. If neither `width` nor `height` are provided, then the image isn't displayed and the message `at least one of width or height must be supplied` is returned.

Another option would be to add `theme-check-disable DeprecatedFilter` to the respective files until we wait for theme check feature/update but disabling this isn't recommended: https://shopify.dev/themes/tools/theme-check/checks/deprecated-filter#disabling-this-check

Reply:
I get you. Delete the paragraph as to not be wrong but leave it undocumented maybe? I believe img_url: 'master' was like this. 

On a different front, I could create a theme-check issue to make a check that enforces the width + height combo instead. This feels like something we might want to enforce "softly" and the docs isn't the right way. A bit like how you can disable ImgLazyLoading when it doesn't make sense. There are valid use cases for asking for the master, it just shouldn't be a default.

Reply:
>Delete the paragraph as to not be wrong but leave it undocumented maybe? I believe img_url: 'master' was like this.

For sure 👌🏼 Or perhaps provide context about what happens if you don't provide a `width` or `height` attribute:

>The intention is that folks shouldn't use it without width or height. While omitting the param does the same thing, it's a bad default because it means the page performance depends on merchant-uploaded-assets.

---

>On a different front, I could create a theme-check issue to make a check that enforces the width + height combo instead. This feels like something we might want to enforce "softly" and the docs isn't the right way. A bit like how you can disable ImgLazyLoading when it doesn't make sense. There are valid use cases for asking for the master, it just shouldn't be a default.

If we could add a comment in the same manner, I think that'd be a great solution for when it doesn't make sense 😄 Please link this PR on that theme-check issue or cc me on it for context. Thanks, @charlespwd

Reply:
Agreed - that'll be the next step, for sure.  Since this PR is just accounting for automatic updates, I think we're good with the current approach but you're 100% right that we should update this to use `image_tag` and ensure that it works as expected.